webui/react/src/pages/ExperimentDetails/CodeViewer/IpynbRenderer.tsx

```diff
@@ -1,20 +1,46 @@
 import NotebookJS from 'notebook';
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+
+import { DetError, ErrorType } from 'shared/utils/error';
+import handleError from 'utils/error';
 
 import 'vendor/monokai.css';
 
 interface Props {
   file: string;
 }
 
+export const parseNotebook = (file: string): string => {
+  try {
+    const json = JSON.parse(file);
+    const notebookJS = NotebookJS.parse(json);
+    return notebookJS.render().outerHTML;
+  } catch (e) {
+    throw new DetError('Unable to parse as Notebook!');
+  }
+};
+
 const JupyterRenderer: React.FC<Props> = React.memo(({ file }) => {
-  return (
-    <div
-      className="ipynb-renderer-root"
-      dangerouslySetInnerHTML={{
-        __html: file && NotebookJS.parse(JSON.parse(file)).render().outerHTML,
-      }}
-    />
+  const [__html, setHTML] = useState<string>();
+
+  useEffect(() => {
+    try {
+      const html = parseNotebook(file);
+      setHTML(html);
+    } catch (error) {
+      handleError(error, {
+        publicMessage: 'Failed to load selected notebook.',
+        publicSubject: 'Unable to parse the selected notebook.',
+        silent: true,
+        type: ErrorType.Input,
+      });
+    }
+  }, [file]);
+
+  return __html ? (
+    <div className="ipynb-renderer-root" dangerouslySetInnerHTML={{ __html }} />
+  ) : (
+    <div>{file}</div>
```

Comment:
That's an interesting point. I was thinking if I'm the user, it would be a bit more useful to me to see the content of the file (if it's visible/able-to-parse) even if it's not a notebook so it might clue me in as to why it's not rendering as a notebook.